packages/@glimmer/opcode-compiler/lib/opcode-builder/helpers/stdlib.ts

```diff
@@ -74,6 +74,11 @@ export function StdAppend(
         op(Op.AppendSafeHTML);
       });
 
+      when(ContentType.TrustedHTML, () => {
+        op(Op.AssertSame);
+        op(Op.AppendHTML);
```

Comment:
Yeah, I see, (mis)using `AppendHTML` here is what caused you to need that unsafe `as string` cast in the other spot. It could probably be DRY-ed up a bit but the easiest thing to do would be just copy the `AppendSafeHTML` opcode into `AppendTrustedHTML` and use that here. 

The more ideal thing is to insert a step in between `Op.AssertSame` and `Op.AppendHTML` that pops the value of the stack, unwraps it, and puts the string value back, that way, `Op.AppendSafeHTML` doesn't need to exist and everything can just use `Op.AppendHTML`

Reply:
I've reused AppendHTML because we don't make modifications to TrustedHTML unlike SafeString. It reflects behavior similar to 
```
foo.innerHTML = trustedHTML
```
we use same DOM methods for 'string' and TrustedHTML, if not casting to string we even wouldn't need this PR.

> The more ideal thing is to insert a step in between Op.AssertSame and Op.AppendHTML that pops the value of the stack, unwraps it, and puts the string value back, that way, Op.AppendSafeHTML doesn't need to exist and everything can just use Op.AppendHTML

that scares me a bit, I don't know engine good enough to make these changes.

Reply:
> So your intention is to actually pass the value all the way through to the DOM functions so it works in whatever stricter environment that would have errored if you do innerHTML(string)

exactly, simply don't break our html we want to render

> SSR/SimpleDOM

It should work with polifill, I also don't think this is concern of Glimmer. Glimmer doesn't create TrustedHTML objects and doesn't try to render them, it simply doesn't modify them.